Libs/DICOM/Core/ctkDICOMDisplayedFieldGeneratorAbstractRule.h

```diff
@@ -1,182 +1,182 @@
+      // the field is not empty, but contain a placeholder string (example: "No description") that means that the field is undefined
+      return true;
+    }
+    // this field is non-empty
+    return false;
+  }
+
+  /// Merge function that only uses the new value if the initial value is empty and vice versa
+  static void mergeExpectSameValue(
+    const QString &fieldName, const QMap<QString, QString> &initialFields, const QMap<QString, QString> &newFields,
+    QMap<QString, QString> &mergedFields, const QMap<QString, QString> &emptyValuesForEachField )
+  {
+    if (isFieldEmpty(fieldName, newFields, emptyValuesForEachField))
+    {
+      // no new value is defined for this value, keep the initial value (if exists)
+      if (!isFieldEmpty(fieldName, initialFields, emptyValuesForEachField))
+      {
+        mergedFields[fieldName]=initialFields[fieldName];
+      }
+      return;
+    }
+    if (isFieldEmpty(fieldName, initialFields, emptyValuesForEachField))
+    {
+      // no initial value is defined for this value, use just the new value (if exists)
+      if (!isFieldEmpty(fieldName, newFields, emptyValuesForEachField))
+      {
+        mergedFields[fieldName]=newFields[fieldName];
+      }
+      return;
+    }
+    // both initial and new value are defined and they are different => just keep using the old value
+    // TODO: log warning here, as this is not expected
+    mergedFields[fieldName]=initialFields[fieldName];
+  }
+
+  /// Merge function that sets merged value as a concatenation of the initial and new values
+  /// Individual values in the concatenation are separated by comma
+  static void mergeConcatenate(
+    const QString &fieldName, const QMap<QString, QString> &initialFields, const QMap<QString, QString> &newFields,
+    QMap<QString, QString> &mergedFields, const QMap<QString, QString> &emptyValuesForEachField )
+  {
+    if (isFieldEmpty(fieldName, newFields, emptyValuesForEachField))
+    {
+      // no new value is defined for this value, keep the initial value (if exists)
+      if (!isFieldEmpty(fieldName, initialFields, emptyValuesForEachField))
+      {
+        mergedFields[fieldName]=initialFields[fieldName];
+      }
+      return;
+    }
+    if (isFieldEmpty(fieldName, initialFields, emptyValuesForEachField))
+    {
+      // no initial value is defined for this value, use just the new value (if exists)
+      if (!isFieldEmpty(fieldName, newFields, emptyValuesForEachField))
+      {
+        mergedFields[fieldName]=newFields[fieldName];
+      }
+      return;
+    }
+    if (initialFields[fieldName].contains(newFields[fieldName]))
+    {
+      // the field is already contained in the list, so no need to add it
+      mergedFields[fieldName]=initialFields[fieldName];
+      return;
+    }
+    // need to concatenate the new value to the initial
+    mergedFields[fieldName]=initialFields[fieldName]+", "+newFields[fieldName];
+  }
+
+protected:
+  ctkDICOMDatabase* DICOMDatabase;
+};
+
+#endif

```

Comment:
why all the lines are marked as changed?

Reply:
Not sure why line endings in some files are converted to Windows style. Thanks for noticing.